pkg/controllers/jivavolume_controller.go

```diff
@@ -199,6 +212,141 @@ func (r *JivaVolumeReconciler) isScaleup(cr *openebsiov1alpha1.JivaVolume) bool
 	return false
 }
 
+func shouldMoveReplicas(cr *openebsiov1alpha1.JivaVolume) bool {
+	if cr.Spec.Policy.Target.ReplicationFactor < 3 {
+		return false
+	}
+	availableReplicas := 0
+	qurom := (cr.Spec.Policy.Target.ReplicationFactor / 2) + 1
+	for _, rep := range cr.Status.ReplicaStatuses {
+		if rep.Mode == "RW" {
+			availableReplicas += 1
+			if availableReplicas == qurom {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (r *JivaVolumeReconciler) moveReplicasForMissingNodes(cr *openebsiov1alpha1.JivaVolume) error {
+
+	// if the volume does not HA replicas in
+	// RW mode skip the process
+	if !shouldMoveReplicas(cr) {
+		return nil
+	}
+
+	var (
+		replicaLabel   = "openebs.io/component=jiva-replica,openebs.io/persistent-volume="
+		nodeAnnotation = "volume.kubernetes.io/selected-node"
+	)
+	labelSelector, err := labels.Parse(
+		replicaLabel + cr.Name)
+	if err != nil {
+		return err
+	}
+	pods := corev1.PodList{}
+	err = r.List(context.TODO(), &pods, &client.ListOptions{
+		Namespace:     cr.Namespace,
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
```

Comment:
here we can optimized by just picking the pending Pod instead of ranging all of them, since we are considering the Node resources in not part of a cluster , all the pod will rescheduled and must be either stuck in pending or scheduled to  new available node